docs/registry/README.md

```diff
@@ -0,0 +1,299 @@
+# Registry Proposal
+
+## Objective
+
+Design an **initial** version of the Registry that can support discoverability of 
+the different event types that can be consumed from the eventing mesh.
+
+## Requirements
+
+Our design revolves around the following core requirements:
+
+1. We should have a Registry per namespace to enforce isolation.
+2. The Registry should contain the event types that can be consumed from
+the eventing mesh. If an event type is not ready for consumption, we should explicitly indicate so.
+3. The event types stored in the Registry should contain all the required information 
+for a consumer to create a Trigger without resorting to some other OOB mechanism.
+
+## Design Ideas
+
+### EventType CRD
+
+We propose having a namespaced EventType CRD. Here is an example of how a CR would look like:
```

Comment:
I have my doubts that a CRD is the right approach for this kind of information. If I understand it correctly, the registry is supposed to be a way for the user to look up event types. The CRs do not describe any desired state on the cluster, and they replicate information the broker got from or through the sources.

Reply:
> If I understand it correctly, the registry is supposed to be a way for the user to look up event types.

Yes, so that users can then do `kubectl get eventtypes -n some-namespace` and get the EventTypes that were instantiated. 
What do you think will be a better approach instead of a CRD?

Reply:
@nachocano I forgot to share also that I've a concern about this being a CRD per EventType for the cases when there are a large number of EventTypes. It is typical for some of our "Event Sources/Enterprise Application" to have around 100+ event types. what are your thoughts around that?

Reply:
@deissnerk  @Abd4llA sorry I may have expressed this wrong. 

There is just one CRD with kind `EventType`. And then EventType CRs will be created that will instantiate that CRD.  There is no CRD per EventType, is just one. 

Similar to Triggers, there is one CRD with kind `Trigger`, and then you can instantiate Trigger CRs with what you want to consume.

Does that make more sense?

Reply:
> around 100+ event types

I guess we will have 100+ EventType CRs in the Registry. Do you think that can be a problem? 
Same thing can happen with Triggers (if you create very specific ones to just do exact matching on type for example). Although probably the typical usage pattern will be to have # triggers << # eventTypes.

Reply:
@nachocano 
The challenge with our event types is that they are delivered using some messaging middleware. The source implementation will not know the event types. It might subscribe to a specific topic that is reserved to one event type and source, but it could also do a wildcard subscription. 
Perhaps we have to distinguish between the events that are available to a Trigger and the events, for which you could configure a Source. 
Of our 100+ events you probably would see a small subset, as most Source definitions wouldn't subscribe to all of them. Consequently creating the Source CRs may be a bigger challenge than the Triggers.

Reply:
@deissnerk makes sense. 
Yeah, for the cases where the Source do not actually know in advance the eventTypes that can be sent (e.g., upon instantiation of a containerSource), the current proposal is to resort to the broker auto-registration policy. But that is kind of a workaround and I'm not entirely convinced. 
We definitively need further discussions on this. Thanks for bringing up that point.